sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/TransformWithStateExec.scala

```diff
@@ -171,3 +175,44 @@ case class TransformWithStateExec(
     }
   }
 }
+
+
+object TransformWithStateExec {
+
+  // Plan logical transformWithState for batch queries
+  def generateSparkPlanForBatchQueries(
+      keyDeserializer: Expression,
+      valueDeserializer: Expression,
+      groupingAttributes: Seq[Attribute],
+      dataAttributes: Seq[Attribute],
+      statefulProcessor: StatefulProcessor[Any, Any, Any],
+      timeoutMode: TimeoutMode,
+      outputMode: OutputMode,
+      outputObjAttr: Attribute,
+      child: SparkPlan): SparkPlan = {
+    val shufflePartitions = child.session.sessionState.conf.numShufflePartitions
+    val statefulOperatorStateInfo = StatefulOperatorStateInfo(
+      Utils.createTempDir().getAbsolutePath,
```

Comment:
We can't expect the path to exist for both driver and executor. If we want to leverage temp dir, the full path should be retrieved from executor.

Also, in flatMapGroupsWithState, we just mapped the batch version of flatMapGroupsWithState to flatMapGroups. I'd guess it's no longer simple as we allow users to initialize multiple states, but would be great if we can fake state instance (or state store implementation) rather than initiating full lifecycle of state store.

Reply:
@HeartSaVioR - the tmp dir creation is based on the `java.io.tmpdir` setting right, which should be OS dependent ? are you saying that it's not reliable if we are running different OS types/versions across driver/executor ? 

unlike FMGWS, we don't actually have a similar batch equivalent like mapGroupsExec that we can use anymore. Implementing fake state store implementation would be fair amount of work I feel (we would also have to ensure that the store supports composite types like ListState, MapState etc in the future) ?

IIUC - the only reason we really need the checkpoint loc for the state store is for keeping track of the committed state across batches (and potentially also within the stateStoreId). Do you think its ok to do either of the following:
- use a local path on the executor (either ways this is a dummy path for batch queries - so I guess should be safe to use)
- skip using the checkpoint loc/pass a None option
- avoid registration for the instance with the state store coordinator ?

Reply:
The code should work with any resource management (standalone, YARN, K8S, etc) with any custom configuration performed by operators/cloud vendors. Also you can override the value on executing JVM process (`-Djava.io.tmpdir`). It is too naive to consider the two nodes for driver and executor to be set up as exactly same.

Maybe I wasn't clear - my ask was mostly about faking state instance (not faking state store implementation - that's mostly rewriting HDFS backed state store provider and we should just use that provider), like abstracting the layer like what we did for FMGWS (GroupStateImpl) and enabling us to provide in-memory value for the state. Yes that may require us to create such fake instance per state type, but it won't be hard as Java/Scala have native types for composite types we are going to support.

If it's non-trivial to inject a layer, I'm OK with spinning up RocksDB state store provider and leveraging it. Pretty sure, it is even better if we can avoid registration with state store coordinator.

Reply:
@HeartSaVioR @anishshri-db 

1. Do we see a usecase where the user would want to read the State files (from DFS) post query?
2. I think implementing a memory based state store is likely a larger effort. For the time being, we can also create the temp directory inside the executor node (inside mapPartitionsWithState), and discard the directory post evaluation (inside completion iterator). I agree though that this is inefficient, and a memory based store would be better in long run.